### ISSUE TITLE
Optional annotations stops working after version 2.16.538 

### ISSUE THREAD

**Issue by kucharzyk** [vojtechhabarta/typescript-generator#440]
With typescript-generator 2.16.538 I use optional annotations.

``` 
<configuration>
  <optionalAnnotations>
    <optionalAnnotation>pl.ics.core.annotations.TypescriptOptional</optionalAnnotation> 
  </optionalAnnotations>
  <optionalProperties>useSpecifiedAnnotations</optionalProperties>
</configuration>
```

My annotation looks like this:

```
package pl.ics.core.annotations

@MustBeDocumented
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class TypescriptOptional
```

It works well with version 2.16.538 but stops working with newer versions.

**Comment by vojtechhabarta** [comment]
I can confirm that there where some changes in version 2.17 regarding annotation handling to support annotations with `TYPE_USE` target.

Could you please give some example class where your annotation is used?

**Comment by kucharzyk** [comment]
Here is sample class which using this annotation:

```
public class BaseDTO implements Serializable {
   @TypescriptOptional
    protected Long id;

    protected Long version;

    @TypescriptOptional
    private String changeReason;

    protected BaseDTO(Long id, Long version) {
        this.id = id;
        this.version = version;
    }

    public BaseDTO(final BaseDTO copy) {
        this.id = copy.getId();
        this.version = copy.getVersion();
        this.changeReason = copy.changeReason;
    }

    public BaseDTO() {
    }

    public Long getId() {
        return this.id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getVersion() {
        return this.version;
    }

    public void setVersion(Long version) {
        this.version = version;
    }

    public String getChangeReason() {
        return changeReason;
    }

    public void setChangeReason(String changeReason) {
        this.changeReason = changeReason;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof BaseDTO)) return false;
        BaseDTO baseDTO = (BaseDTO) o;
        return Objects.equals(getId(), baseDTO.getId()) &&
            Objects.equals(getVersion(), baseDTO.getVersion());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getId(), getVersion());
    }
}

```

**Comment by kucharzyk** [comment]
Another example:

```
public class SearchDTO {

    private Integer year;
    // ...
    @TypescriptOptional
    private Long selectedId;

   // getters, setters, no-arg constructor
}
```

**Comment by vojtechhabarta** [comment]
If you move `@TypescriptOptional` annotataion from `id` field to `getId()` getter it should work. That's because bean properties are defined by **public** members. So here `id` property is defined by `getId()` getter, not `id` field.

I am thinking if it is worth returning old behavior back because it depends on exact name convention. When I have for example `getId()` getter and `id` field annotated as optional and I rename it to `dtoId` it suddenly doesn't work which might be surprising. In other words: by changing internal implementation I am changing also public interface.

**Comment by dkowis** [comment]
We use this exact style when the class is generated via Lombok. So the annotation is stuck on the field, and the getters/setters are automagically generated, so they're always going to match 100% of the time.
The methods are generated at build time to javabean conventions.

**Comment by dkowis** [comment]
We found a quick workaround when using lombok `@Data`. For the annotation: 
```java
@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface TypescriptOptional {
}
```
You can annotate each field to be optional with: 
```java
    @Getter(onMethod_={@TypescriptOptional})
    private CalculationCountryCode countryCode;
```
This will then generate proper typescript:
```typescript
    countryCode?: CalculationCountryCode;
```

**Comment by vojtechhabarta** [comment]
@dkowis many thanks for sharing the solution for lombok.

I also checked Immutables framework but in its case it is other way around (writing getters, generating implementation fields).

**Comment by vojtechhabarta** [comment]
In ac7f82143ca1aa42d8206ae4c31ee2b010bda940 commit I changed how annotations are read. Now Jackson's `getAnnotation` method is called again (as it was before 2.17 changes). So when some property has `private` field and `public` getter annotations from both of these members are considered.

**Comment by vojtechhabarta** [comment]
Released in [v2.21.588](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v2.21.588).
